library/aes.c

```diff
@@ -1446,44 +1446,66 @@ int mbedtls_aes_self_test( int verbose )
             mbedtls_printf( "  AES-CTR-128 (%s): ",
                              ( v == MBEDTLS_AES_DECRYPT ) ? "dec" : "enc" );
 
-        memcpy( nonce_counter, aes_test_ctr_nonce_counter[u], 16 );
         memcpy( key, aes_test_ctr_key[u], 16 );
 
-        offset = 0;
         mbedtls_aes_setkey_enc( &ctx, key, 128 );
 
         if( v == MBEDTLS_AES_DECRYPT )
         {
             len = aes_test_ctr_len[u];
-            memcpy( buf, aes_test_ctr_ct[u], len );
-
-            mbedtls_aes_crypt_ctr( &ctx, len, &offset, nonce_counter, stream_block,
-                           buf, buf );
-
-            if( memcmp( buf, aes_test_ctr_pt[u], len ) != 0 )
+            tst_offset = 0;
+            while ( tst_offset < 16 )
             {
-                if( verbose != 0 )
-                    mbedtls_printf( "failed\n" );
-
-                ret = 1;
-                goto exit;
+                memcpy( buf, aes_test_ctr_ct[u], len );
+                memcpy( nonce_counter, aes_test_ctr_nonce_counter[u], 16 );
+                offset = 0;
```

Comment:
Shouldn't ```offset``` i.e. ```nc_off``` be set outside the loop? As iterations intend to resume ciphering in stream ```stream_block```?

Reply:
The `nc_off` is an IN\OUT parameter, and gets modified to the offset of the last block.  I want every iteration to start from the beginning of the stream ( offset 0 )
Perhaps the naming is not good.